### ISSUE TITLE
[Feature Request] Upstream the zsh completions

### ISSUE THREAD

**Issue by NicolaSmaniotto** [iamkroot/trakt-scrobbler#173]
It would be nice to upstream the zsh completions file present [in the AUR](https://aur.archlinux.org/cgit/aur.git/tree/trakts.zsh?h=trakt-scrobbler), to make it available for other distributions and have a place to maintain it in case something changes in the future.

**Comment by iamkroot** [comment]
Thanks for creating the issue! I've been lazy about this, and I looked into it a bit.

We use [Cleo](http://cleo.readthedocs.io/) under the hood to create the CLI, and turns out, it has built-in support for `bash`, `zsh` and `fish` [autocompletions](https://github.com/sdispater/cleo/tree/0.8.1#autocompletion)).

You just need to follow the steps from the link and the completions will be generated!
@ahmubashshir Maybe you can update the PKGBUILD to do this instead of pre-generating them?

**Comment by ahmubashshir** [comment]
Okay, I'll look into it.
	Mubashshir

**Comment by ahmubashshir** [comment]
BTW, 'plex' subcommand description is probably incorrect,

```
$ trakts help
...
auth    Runs the authentication flow for trakt.tv
...
plex    Runs the authentication flow for trakt.tv
...
```
	Mubashshir

**Comment by NicolaSmaniotto** [comment]
PR #174 is a good thing, thanks @ahmubashshir. I'd have opened it myself, but I'm not familiar with the AUR's licensing.

As soon as I have some time I'll play a bit with cleo's autogenerated completions, those would solve every issue.

**Comment by NicolaSmaniotto** [comment]
> The autogenerated completion for zsh is a bit buggy, it autocompletes path even if the command will not accept the path.
> 
> ![image](https://user-images.githubusercontent.com/24864366/151842530-8177c33b-8b10-470d-80cd-d5d9bd6aba39.png)

This seems to be sdispater/cleo#85, which would be solved by merging sdispater/cleo#86.

**Comment by iamkroot** [comment]
Yeah, the cleo autogenerated completions don't seem to support sub-commands. It is only providing top-level command completions.
@ahmubashshir's completions are definitely more _complete_ (pardon the pun :P)

**Comment by NicolaSmaniotto** [comment]
My thought exactly. It's sad, but cleo seems to be pretty limited to what the dev needs it to do for poetry.

The issue to complete subcommands was opened as sdispater/cleo#84 in september 2020, with ho response yet.

I patched my cleo install with sdispater/cleo#86, and it solves the "bad set [...] array" problem, but that too was opened in september 2020 and still not merged.

I agree that the best approach for now is #174, at least until cleo adds the missing features.
